lib/phantomjs.js

```diff
@@ -61,3 +63,34 @@ if (exports.path) {
     // We did our best. Likely because phantomjs was already installed.
   }
 }
+
+/**
+ * Executes a script or just runs PhantomJS
+ */
+exports.exec = function () {
+  var args = Array.prototype.slice.call(arguments)
+  return spawn(exports.path, args)
+}
+
+/**
+ * Runs PhantomJS with provided options
+ * @example
+ * // handy with WebDriver
+ * phantomjs.run('--webdriver=4444').then(program => {
+ *   // do something
+ *   program.kill()
+ * })
+ * @returns {Promise} the process of PhantomJS
+ */
+exports.run = function () {
+  var program = exports.exec.apply(null, arguments)
```

Comment:
oops, meant to say "if the process _exited_ before the promise resolved". if it's not easy to do, don't worry about it.